File: src/types/video.ts
import { CSSProperties } from "react";

type Credit = {
  name: string | string[];
  role: string;
}

export type VideoMetadata = {
  title: string;
  description?: string;
  weight?: number; // weight of how much to show in the header.
  useboxartaspreview?: boolean;
  boxart: {
    title: string;
    caption: string;
    img: string;
    video?: string;
  };
  mobileBoxart?: {
    title: string;
    caption: string;
    img: string;
    video?: string;
  };
  links?: {
    youtube?: string;
    vimeo?: string;
  }
  credits?: Credit[];
  url: string;
  date: string;
  img?: string;
  style?: CSSProperties;
}

File: backend/internal/components/video/model/video.go
package model

type Video struct {
	VideoId     int64    `json:"videoId"`
	Title       string   `json:"title"`
	Description *string  `json:"description"`
	Boxart      *Boxart  `json:"boxart,omitempty"`
	Weight      int      `json:"weight"`
	Credits     []Credit `json:"credits,omitempty"`
	Links       []Link   `json:"links,omitempty"`
	VideoType   string   `json:"videoType"`
	CreatedAt   string   `json:"createdAt"`
	UpdatedAt   string   `json:"updatedAt"`
}
